test/unit/lib/plugins/aws/package/compile/events/alb/index.test.js

```diff
@@ -116,4 +130,64 @@ describe('lib/plugins/aws/package/compile/eents/alb/index.test.js', () => {
       );
     });
   });
+
+  describe('alb rule conditions', () => {
+    it('should support rule without path', () => {
+      const albListenerRuleLogicalId = naming.getAlbListenerRuleLogicalId(
+        'fnConditionsHostOnly',
+        4
+      );
+      const rule = cfResources[albListenerRuleLogicalId];
+
+      expect(rule.Type).to.equal('AWS::ElasticLoadBalancingV2::ListenerRule');
+      expect(rule.Properties.Conditions).to.have.length(1);
+      expect(rule.Properties.Conditions[0].Field).to.equal('host-header');
+      expect(rule.Properties.Conditions[0].Values).to.have.length(1);
+      expect(rule.Properties.Conditions[0].Values[0]).to.equal('example.com');
+    });
+
+    it('should should support rule with path', () => {
+      const albListenerRuleLogicalId = naming.getAlbListenerRuleLogicalId(
+        'fnConditionsPathOnly',
+        5
+      );
+      const rule = cfResources[albListenerRuleLogicalId];
+
+      expect(rule.Type).to.equal('AWS::ElasticLoadBalancingV2::ListenerRule');
+      expect(rule.Properties.Conditions).to.have.length(1);
+      expect(rule.Properties.Conditions[0].Field).to.equal('path-pattern');
+      expect(rule.Properties.Conditions[0].Values).to.have.length(1);
+      expect(rule.Properties.Conditions[0].Values[0]).to.equal('/');
+    });
+
+    it('should fail validation if no conditions are set', () => {
+      return runServerless({
+        fixture: 'function',
+        cliArgs: ['package'],
+        configExt: {
+          functions: {
+            fnConditionsHostOnly: {
+              handler: 'index.handler',
+              events: [
+                {
+                  alb: {
+                    ...baseEventConfig,
+                    priority: 1,
+                    conditions: {},
+                  },
+                },
+              ],
+            },
+          },
+        },
+      }).then(
+        () => Promise.reject('Expected to fail'),
+        r =>
+          assert.equal(
+            r,
+            'ServerlessError: At least 1 condition for alb event must be set in function "fnConditionsHostOnly".'
```

Comment:
Let's confirm on error code instead of message (Message is subject to changes, and changing it should not affect tests).

Note: `code` is accepted as second argument in `ServerlessError` constructor

Reply:
@medikoo Any suggestions on what to use for the error code?

Reply:
Maybe `ALB_NO_CONDITIONS` (?)